src/transformers/models/llava_next/image_processing_llava_next.py

```diff
@@ -0,0 +1,709 @@
+
+    Returns:
+        list: A list of np.array representing the patches.
+    """
+    patches = []
+    height, width = image.shape[:-1] if input_data_format == ChannelDimension.LAST else image.shape[1:]
+    for i in range(0, height, patch_size):
+        for j in range(0, width, patch_size):
+            if input_data_format == ChannelDimension.LAST:
+                patch = image[i : i + patch_size, j : j + patch_size]
+            else:
+                patch = image[:, i : i + patch_size, j : j + patch_size]
+            patches.append(patch)
+
+    return patches
+
+
+def expand_to_square(image: np.array, background_color, input_data_format) -> np.array:
+    """
+    Expands an image to a square by adding a background color.
+    """
+
+    height, width = get_image_size(image, channel_dim=input_data_format)
+    if width == height:
+        return image
+    elif width > height:
+        result = np.ones((width, width, image.shape[2]), dtype=image.dtype) * background_color
+        result[(width - height) // 2 : (width - height) // 2 + height, :] = image
+        return result
+    else:
+        result = np.ones((height, height, image.shape[2]), dtype=image.dtype) * background_color
+        result[:, (height - width) // 2 : (height - width) // 2 + width] = image
+        return result
+
+
+class LlavaNextImageProcessor(BaseImageProcessor):
+    r"""
+    Constructs a LLaVa-NeXT image processor. Based on [`CLIPImageProcessor`] with incorporation of additional techniques
+    for processing high resolution images as explained in the [LLaVa paper](https://arxiv.org/abs/2310.03744).
+
+    Args:
+        aspect_ratio_setting (`str`, *optional*, defaults to `"anyres"`):
+            The aspect ratio setting to use. Can be "clip" (as in CLIP), "pad" (as in LLaVa 1.5) or "anyres" (as in LLaVa 1.6).
+            an be overridden by `aspect_ratio_setting` in the `preprocess` method.
+        do_resize (`bool`, *optional*, defaults to `True`):
+            Whether to resize the image's (height, width) dimensions to the specified `size`. Can be overridden by
+            `do_resize` in the `preprocess` method.
+        size (`Dict[str, int]` *optional*, defaults to `{"shortest_edge": 224}`):
+            Size of the image after resizing. The shortest edge of the image is resized to size["shortest_edge"], with
+            the longest edge resized to keep the input aspect ratio. Can be overridden by `size` in the `preprocess`
+            method.
+        image_grid_pinpoints (`List` *optional*, defaults to `[[672, 336], [336, 672], [672, 672], [336, 1008], [1008, 336]]`):
+            A list of possible resolutions to use for processing high resolution images. The best resolution is selected
+            based on the original size of the image. Can be overridden by `image_grid_pinpoints` in the `preprocess`
+            method.
+        resample (`PILImageResampling`, *optional*, defaults to `Resampling.BICUBIC`):
```

Comment:
Double checking this matches the original implementation

Reply:
Yes, original implementation also uses BICUBIC interpolation, which is asserted in the conversion script [here](https://github.com/NielsRogge/transformers/blob/add_llava_1_6_bis/src/transformers/models/llava_next/convert_llava_next_weights_to_hf.py#L174)